Update Theme to Latest Hugo Version

andrew-aiken opened the pull request:
On the newest version of hugo the `resources.ToCSS` and `resources.PostCSS` are deprecated.

#### Overview of changes:
- Update replicated function calls
- Update packages in example site

#### Replication of Issue
```bash
> hugo server
Start building sites …
hugo v0.143.1+extended+withdeploy darwin/arm64 BuildDate=2025-02-04T08:57:38Z VendorInfo=brew

ERROR deprecated: resources.ToCSS was deprecated in Hugo v0.128.0 and subsequently removed. Use css.Sass instead.
ERROR deprecated: resources.PostCSS was deprecated in Hugo v0.128.0 and subsequently removed. Use css.PostCSS instead.
```

andrew-aiken committed:
Update deprecated css function

Signed-off-by: Andrew Aiken <andrew.aiken@machinemetrics.com>

andrew-aiken committed:
Update example packages

Signed-off-by: Andrew Aiken <andrew.aiken@machinemetrics.com>

andrew-aiken committed:
Add back external modules and update lock

Signed-off-by: Andrew Aiken <andrew.aiken@machinemetrics.com>

andrew-aiken committed:
Update netlify node and hugo versions

Signed-off-by: Andrew Aiken <andrew.aiken@machinemetrics.com>

cjyabraham reviewed (changes requested):
Thanks for this contribution. 

When I compare [the homepage of your dev site](https://deploy-preview-65--dot-org-hugo-theme-demo-thetwopct.netlify.app/) with [the homepage of the sample site](https://main--dot-org-hugo-theme-demo.netlify.app/), there's a difference towards the bottom. The "Example Page" shows just an excerpt on the sample site but your dev site is showing formatted contents of the page.

The blog listing also has the same difference from [your dev site](https://deploy-preview-65--dot-org-hugo-theme-demo-thetwopct.netlify.app/blog/) to [sample site](https://main--dot-org-hugo-theme-demo.netlify.app/blog/) in how excerpts are handled. I think I like how the sample site has it as it's more compact. 

Looks like both of the above issues are caused by a change in how Hugo handles the `{{ .Summary }}` template tag. Maybe change it to `{{ .Summary | plainify }}`?

andrew-aiken committed:
Remove post summaries and set max length

Signed-off-by: Andrew Aiken <andrew.aiken@machinemetrics.com>

andrew-aiken commented:
Thanks for the catch

Updated to fix the summary being included.

I also added `summaryLength` of 50 because without it the summary would not automatically end. Alternatively we could add `<!--more-->` but then it would have to be added to each post.

cjyabraham reviewed (approved):
Looks good. Thanks again for this!
